### PR TITLE
Update code for changes to BoB robotics API

### PR DESCRIPTION
**Pull request by alexdewar**
Hey,

It looks like I've broken this code by changing the BoB robotics API in a couple of places and @tenxlenx wanted to reuse some of it for his work. So I've fixed up the code in a couple of places and pegged it to a specific version of BoB robotics by adding a submodule like we've done elsewhere. (If you don't want to use the submodule you can just do ``USE_LOCAL_BOB_ROBOTICS=1 cmake ..``.)

### CODE REVIEW

File: memory.cc [hunk 1]
```diff
@@ -61,9 +61,9 @@ std::vector<float> PerfectMemory::calculateRIDF(const cv::Mat &snapshot) const
     std::vector<float> ridf(getImageSize().width, std::numeric_limits<float>::max());
 
     // Loop through all snapshots
-    for(const auto &memDifferences : allDifferences) {
-        for(int c = 0; c < getImageSize().width; c++) {
-            ridf[c] = std::min(ridf[c], memDifferences[c]);
+    for (int i = 0; i < allDifferences.rows(); i++) {
```
Comment (neworderofjamie) [inline memory.cc:64]: is this not a vector of vectors anymore?

**Review by neworderofjamie — APPROVED** [review]
Thanks for this - would have been totally happy to clean up my own mess here 😄

File: memory.cc (hunk 1 above)
Comment (alexdewar) [reply memory.cc:64]: Nope, I changed it to be an Eigen matrix instead. We're using Eigen for some of the other navigation code anyway, so I thought we might as well use it here. It just makes it easier in terms of resizing etc. and I think it's a tiny bit faster too...

Comment (neworderofjamie) [reply memory.cc:64]: 👍

**Pull request merged**